google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +135,33 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    Throwable throwable = exs;
+    while (throwable != null) {
+      if (throwable.getMessage().contains("Connection closed prematurely")
+          || throwable.getCause() instanceof javax.net.ssl.SSLException
+          || throwable.getCause() instanceof java.io.EOFException
+          || throwable.getCause() instanceof java.net.SocketException
+          || throwable.getCause() instanceof java.net.SocketTimeoutException) {
+        return true;
+      }
+      throwable = throwable.getCause();
+    }
+    return false;
+  }
+
   private void sleepForAttempt(int attempt) {
+    long delay = 1000;
+    long maxDelay = 120_000;
+    // exponential backoff
+    for (int i=0; i<attempt; i++) {
+      delay *= 2;
```

Comment:
Just a note, if you choose to keep 2 as a backoff factor. For the current implementation, if the goal is to achieve `delay = 1000 * 2 ^ attempt`, there is no need to have a loop:
`delay = 1000 * (2 << attempt)`;